velox/connectors/clp/search_lib/CMakeLists.txt

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set(CLP_EXTERNAL_BINARY_DIR ${CMAKE_BINARY_DIR}/external/clp)
+add_subdirectory(${clp_SOURCE_DIR}/components/core/src/clp/string_utils
+                 ${CLP_EXTERNAL_BINARY_DIR}/string_utils)
+set(YSTDLIB_CPP_BUILD_TESTING OFF)
+add_subdirectory(${clp_SOURCE_DIR}/components/core/submodules/ystdlib-cpp
+                 ${CLP_EXTERNAL_BINARY_DIR}/ystdlib-cpp EXCLUDE_FROM_ALL)
+
+string(LENGTH "${CMAKE_SOURCE_DIR}/" SOURCE_PATH_SIZE)
+
+antlr_target(
+  KqlParser
+  ${CLP_SRC_DIR}/clp_s/search/kql/Kql.g4
+  LEXER
+  PARSER
+  VISITOR
+  PACKAGE
+  kql)
+
+set(CLP_SRC_FILES
+    ${ANTLR_KqlParser_CXX_OUTPUTS}
+    ${CLP_SRC_DIR}/clp_s/ArchiveReader.cpp
+    ${CLP_SRC_DIR}/clp_s/ArchiveReaderAdaptor.cpp
+    ${CLP_SRC_DIR}/clp_s/ColumnReader.cpp
```

Comment:
Can you explain what's happening here? This seems like vendoring with extra steps. (steps that add a bunch of dependencies...)

Reply:
Sorry about it. We are doing some refactoring work and fixing CI failures. I'll get back to you when this PR is ready.